README.md

```diff
@@ -1 +1,218 @@
+
+## Objectives
+
+1. Explain the event emitter pattern
+1. Implement an event emitter (inheriting from the base class with util.inherits)
+1. Describe about some of the event from http module
+1. Describe how to parse the information from the response
+
+## Observer Pattern
+
+The observer pattern is common in most of the programming languages. In a nutshell, you have one object which listens to events. Those events can be triggered in multiple places at a different times. By creating event listeners, developers can add some logic which they want to execute on a certain event. For example, you have a button on a web page. You want to see an image when the button is clicked. You can set up an event `onClick` and implement event listener with some browser JavaScript which will show the image.
+
+In Node, we use Event Emitter class to implement an observer. 
+
+## Event Emitter
+
+The Event Emitter object can listen to events. To create it, import the core module `events` and instantiate the object with `new`:
+
+
+```js
+var events  = require('events')
+var emitter = new events.EventEmitter()
+```
+
+To illustrate how Event Emitter works, we will create an object which can listen to the event `knock`. Go ahead and create a file `knock-knock.js`. Create the `emitter` object as shown above, and implement two event listeners:
+
+
+```js
+emitter.on('knock', function() {
+    console.log('Who\'s there?')
+})
+
+emitter.on('knock', function() {
+    console.log('Go away!')
+})
+```
+
+```js
+emitter.emit('knock')
+```
+
+Run your newly created Node script with this command:
+
+```
+node knock-knock.js
+```
+
+You should see the output produced by two event listeners in the order in which they were defined because the event `knock` was triggered by the last statement:
+
+```
+Who's there?
+Go away!
+```
+
+Just for fun, add another `emitter.emit('knock')`. There would be there output:
+
+```
+Who's there?
+Go away!
+Who's there?
+Go away!
+```
+
+To summarize, we defined two different event listeners for the same arbitrary event `knock`. Then we triggered the event twice. 
+
+Additional methods include:
+
+* `emitter.listeners(eventName)`: List the event listeners associated with the `eventName` event
+* `emitter.once(eventName, listener)`: Create event listener for `eventName` and execute it just one time. This is similar to `.on()` but fires only once.
+* `emitter.removeListener(eventName, listener)`: Remove event listener for `eventName`
+
+In our knock-knock example, we used the observer object `event`. It's not very interesting because it didn't do much. What's more interesting and useful is that we can extend Event Emitter properties and methods into ANY OBJECT/class. There's this convenient function `inherits()` from the core `util` module. It takes a child (`fancyChildObjectWithEvents`) and parent objects, and makes the child inherit from the parent (Event Emitter).
```

Comment:
The first sentence could be really confusing for a beginner -- could we unpack that we're using the `EventEmitter` object from the `events` module in the Node.js standard library?